libmamba/tests/src/util/test_iterator.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <doctest/doctest.h>
+
+#include "mamba/util/iterator.hpp"
+
+using namespace mamba::util;
+
+namespace
+{
+    bool greater_than_10(int i)
+    {
+        return i > 10;
+    }
+
+    bool second_greater_than_10(const std::pair<std::string, int>& p)
+    {
+        return p.second > 10;
+    }
+
+    auto get_test_data()
```

Comment:
Why make a tuple of these? Why not just set them as constant data? They can be easilly copied in the tests suites. Or wrapped in a srtuct, to not have to re-declare names in tests, just declar an object of that struct type and you're good to go.

Reply:
Indeed, I started with a struct and then changed for this and I cannot remember why :sweat_smile: Let's go back to the struct.